src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -401,23 +401,38 @@ void AppHost::LastTabClosed(const winrt::Windows::Foundation::IInspectable& /*se
         _HideNotificationIconRequested();
     }
 
+    // We don't want to try to save layouts if we are about to close
+    _getWindowLayoutThrottler.reset();
+    _windowManager.GetWindowLayoutRequested(_GetWindowLayoutRequestedToken);
+
     _window->Close();
 }
 
 LaunchPosition AppHost::_GetWindowLaunchPosition()
 {
-    // Get the position of the current window. This includes the
-    // non-client already.
-    const auto window = _window->GetWindowRect();
+    LaunchPosition pos{};
+    // If we started saving before closing, but didn't resume the event handler
+    // until after _window might be a nullptr.
+    if (!_window)
+    {
+        return pos;
+    }
 
-    const auto dpi = _window->GetCurrentDpi();
-    const auto nonClientArea = _window->GetNonClientFrame(dpi);
+    try
+    {
+        // Get the position of the current window. This includes the
+        // non-client already.
+        const auto window = _window->GetWindowRect();
 
-    // The nonClientArea adjustment is negative, so subtract that out.
-    // This way we save the user-visible location of the terminal.
-    LaunchPosition pos{};
-    pos.X = window.left - nonClientArea.left;
-    pos.Y = window.top;
+        const auto dpi = _window->GetCurrentDpi();
+        const auto nonClientArea = _window->GetNonClientFrame(dpi);
+
+        // The nonClientArea adjustment is negative, so subtract that out.
+        // This way we save the user-visible location of the terminal.
+        pos.X = window.left - nonClientArea.left;
+        pos.Y = window.top;
+    }
+    CATCH_LOG();

```

Comment:
It felt safer to also just put the try/catch here since this function is called in the `CloseRequested` and `WindowCloseButtonClicked` event handlers as well, and might be used elsewhere in the future.